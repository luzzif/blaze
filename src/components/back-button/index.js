import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { ArrowBack } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicBackButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <ArrowBack />
    </IconButton>
);

BasicBackButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const BackButton = enhanceComponent(BasicBackButton);
