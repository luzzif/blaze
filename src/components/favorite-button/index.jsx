import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { Favorite } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicFavoriteButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <Favorite />
    </IconButton>
);

BasicFavoriteButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const FavoriteButton = enhanceComponent(BasicFavoriteButton);
