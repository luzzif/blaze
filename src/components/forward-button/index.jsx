import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { ArrowForward } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicForwardButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <ArrowForward />
    </IconButton>
);

BasicForwardButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const ForwardButton = enhanceComponent(BasicForwardButton);
