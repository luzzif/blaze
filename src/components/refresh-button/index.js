import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { Refresh } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicRefreshButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <Refresh />
    </IconButton>
);

BasicRefreshButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const RefreshButton = enhanceComponent(BasicRefreshButton);
