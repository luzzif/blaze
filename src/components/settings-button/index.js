import React from "react";
import PropTypes from "prop-types";
import { IconButton, withStyles } from "@material-ui/core";
import { Menu } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";

const BasicSettingsButton = ({ classes, ...rest }) => (
    <IconButton {...rest} classes={{ root: classes.root }}>
        <Menu />
    </IconButton>
);

BasicSettingsButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const SettingsButton = enhanceComponent(BasicSettingsButton);
