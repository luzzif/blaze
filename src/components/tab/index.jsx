import React from "react";
import { Card, CardContent } from "@material-ui/core";

export const Tab = props => (
    <Card>
        <CardContent />
    </Card>
);
