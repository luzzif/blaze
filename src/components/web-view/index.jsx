import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core";
import { styles } from "./styles";
import { compose } from "redux";

const BasicWebView = ({ classes, ...rest }) => (
    <webview className={classes.root} src="http://www.github.com" />
);

BasicWebView.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const WebView = enhanceComponent(BasicWebView);
