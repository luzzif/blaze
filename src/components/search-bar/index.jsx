import React from "react";
import PropTypes from "prop-types";
import { Card, CardContent, withStyles, Grid } from "@material-ui/core";
import { compose } from "redux";
import { styles } from "./styles";
import { FavoriteButton } from "../favorite-button";

const BasicSearchBar = ({ classes, ...rest }) => (
    <Card {...rest} classes={{ root: classes.root }}>
        <CardContent classes={{ root: classes.content }}>
            <div className={classes.container}>
                <input className={classes.input} type="text" />
                <FavoriteButton />
            </div>
        </CardContent>
    </Card>
);

BasicSearchBar.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const SearchBar = enhanceComponent(BasicSearchBar);
