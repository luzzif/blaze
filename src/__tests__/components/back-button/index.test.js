import React from "react";
import { ArrowBack } from "@material-ui/icons";
import { BackButton } from "./../../../components/back-button";
import TestRenderer from "react-test-renderer";

describe("back button", () => {
    let testInstance = null;

    beforeEach(() => {
        testInstance = TestRenderer.create(<BackButton />);
    });

    it("renders correctly", () => {
        expect(testInstance.toJSON().type).toBe("button");
        expect(testInstance).toMatchSnapshot();
    });

    it("should be clickable", () => {
        expect(testInstance.root.findByType("button").props.disabled).toBe(false);
    });

    it("should contain a arrow back icon", () => {
        expect(testInstance.root.findByType(ArrowBack)).toBeTruthy();
        expect(testInstance).toMatchSnapshot();
    });
});
