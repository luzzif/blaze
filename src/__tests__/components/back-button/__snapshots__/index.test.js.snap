exports[`back button renders correctly 1`] = `
<button
  className="MuiButtonBase-root-8 MuiIconButton-root-2 BasicBackButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button">
  <span
    className="MuiIconButton-label-7">
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-11"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24">
      <path
        d="M0 0h24v24H0z"
        fill="none" />
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-18" />
</button>
`;

exports[`back button should be clickable 1`] = `
<button
  className="MuiButtonBase-root-8 MuiIconButton-root-2 BasicBackButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button">
  <span
    className="MuiIconButton-label-7">
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-11"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24">
      <path
        d="M0 0h24v24H0z"
        fill="none" />
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-18" />
</button>
`;

exports[`back button should contain a arrow back icon 1`] = `
<button
  className="MuiButtonBase-root-8 MuiIconButton-root-2 BasicBackButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button">
  <span
    className="MuiIconButton-label-7">
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-11"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24">
      <path
        d="M0 0h24v24H0z"
        fill="none" />
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z" />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-18" />
</button>
`;
